### PR TITLE
[docs] Add a reference guide custom build configs

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Closes ENG-7735

# How

This PR:
- adds a step-by-step on how to create a custom build (uses running tests as a workflow example).
- the link is hidden from search and is not part of the navigation structure (yet) since this feature is in active development. Adding it to docs, we can share the https://docs.expo.dev/build-reference/custom-build-config/ link for reference. (let me know @dsokal, if this is okay, otherwise we can look into other ways to share this)

# Test Plan

Changes can be tested locally by visiting http://localhost:3002/build-reference/custom-build-config/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/custom-build-config.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+import ImageSpotlight from '~/components/plugins/ImageSpotlight';
+import { BoxLink } from '~/ui/components/BoxLink';
+import { GithubIcon } from '@expo/styleguide-icons';
+
+Custom workflows allow running tests on the cloud, adding instructions for native platforms, resigning a build to share with your team or multiple test devices, and so on. With EAS Build, you can create build configs to customize the process for adding your workflows. It helps saving time especially when working teams.
+
+This guide shows how to create and use a custom build config with EAS. It follows a scenario where you create a custom config to run tests on EAS Build. You can also use the defined steps in other scenarios.
```
Comment (dsokal) [inline docs/pages/build-reference/custom-build-config.mdx:15]:
```suggestion
This guide shows how to create and use a custom build config with EAS. It follows a scenario where you create a custom workflow to run tests on EAS Build. You can also use the defined steps in other scenarios.
```

File: docs/pages/build-reference/custom-build-config.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          npm test
+```
+
+</Step>
+
+<Step label="3">
+
+## Add `config` property in eas.json
+
+To use the custom build config, you must add the `config` property in **eas.json**. It contains the build config file's name as the value you want to run on the EAS.
```
Comment (dsokal) [inline docs/pages/build-reference/custom-build-config.mdx:94]:
```suggestion
To use the custom build config, you must add the `config` property in **eas.json**. It contains the build config filename as the value you want to run on the EAS.
```

File: docs/pages/build-reference/custom-build-config.mdx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+</Step>
+
+## Summary
+
+- Build configs are customizable by adding steps you want to run on the EAS service as an extension to your existing regular builds.
+- When the **eas.json** file contains the `config` property, credentials such as the keystore setup for Android and distribution certificate for iOS are skipped. This is not implemented yet because both the keystore and distribution certificate are unavailable on the server side.
+- If an existing project already has credentials (keystore and distribution certificate), they remain unchanged.
+- If your regular builds have [EAS secrets](/build-reference/variables/#adding-secrets-with-eas-cli), they are also available for custom builds.
+
+Also, check out the **example repository** for a more detailed example:
```
Comment (dsokal) [inline docs/pages/build-reference/custom-build-config.mdx:136]:
```suggestion
Also, check out the **example repository** for more detailed examples:
```

**Review by dsokal — APPROVED** [review]
Thank you 💯

File: docs/pages/build-reference/custom-build-config.mdx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+hideFromSearch: true
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+import ImageSpotlight from '~/components/plugins/ImageSpotlight';
+import { BoxLink } from '~/ui/components/BoxLink';
+import { GithubIcon } from '@expo/styleguide-icons';
+
+Custom workflows allow running tests on the cloud, adding instructions for native platforms, resigning a build to share with your team or multiple test devices, and so on. With EAS Build, you can create build configs to customize the process for adding your workflows. It helps saving time especially when working teams.
```
[the remark this answers is not in the document]
Comment (amandeepmittal) [reply docs/pages/build-reference/custom-build-config.mdx:13]: My bad and you are right. It should be "when working in teams."

Comment (amandeepmittal) [inline docs/pages/build-reference/custom-build-config.mdx:13]:
```suggestion
Custom workflows allow running tests on the cloud, adding instructions for native platforms, resigning a build to share with your team or multiple test devices, and so on. With EAS Build, you can create build configs to customize the process for adding your workflows. It helps saving time especially when working in teams.
```

**Pull request merged**